RELEASE/scripts/autoscend/iotms/mr2023.ash

```diff
@@ -868,6 +878,320 @@ void auto_handleJillOfAllTrades()
+			return true;
+		}
+		else
+		{
+			auto_log_debug("Attempted to get citizen of a zone buff " + goal + " however we failed.");
+			return false;
+		}
+	}
+	else
+	{
+		auto_log_debug("Attempted to get citizen of a zone buff " + goal + " however we failed.");
+		return false;
+	}
+}
+
+boolean auto_getCitizenZone(string goal)
+{
+	familiar eagle = $familiar[Patriotic Eagle];
+	//zones are approximately organized by autoscend level quest structure
+	boolean[location] meatZones = $locations[The Battlefield (Frat Uniform), The Hidden Hospital, The Haunted Bathroom, The Castle in the Clouds in the Sky (Basement),
+	Lair of the Ninja Snowmen, The Defiled Cranny, The Laugh Floor, The Batrat and Ratbat Burrow, The Sleazy Back Alley];
+	boolean[location] itemZones = $locations[A Massive Ziggurat, The Haunted Laundry Room, Whitey's Grove, The Icy Peak, Itznotyerzitz Mine,
+	The Dark Heart of the Woods, The Hidden Temple, The Haunted Library, The Bat Hole Entrance, Noob Cave];
+	boolean[location] initZones = $locations[The Feeding Chamber, An Unusually Quiet Barroom Brawl, An Overgrown Shrine (Northeast),
+	Oil Peak, Cobb's Knob Kitchens, The VERY Unquiet Garves, The Haunted Kitchen];
+	//mp zones are organized by 20-30 mp regen then 10-15 mp regen and then approximately autoscend level quest structure
+	boolean[location] mpZones = $locations[Sonofa Beach, The Themthar Hills, The Upper Chamber, Inside the Palindome, An Overgrown Shrine (Northwest), A-boo Peak, Hippy Camp,
+	Megalo-City, Shadow Rift, Vanya's Castle, The Hatching Chamber, Wartime Hippy Camp (Frat Disguise), Frat House, The Middle Chamber, The Black Forest,
+	The Haunted Ballroom, The Red Zeppelin, An Overgrown Shrine (Southwest), The Hidden Park, Twin Peak, The Smut Orc Logging Camp, The Daily Dungeon, The Spooky Forest];
+	boolean[location] specZones = $locations[The Outskirts of Cobb\'s Knob];
+	string activeCitZoneMod = activeCitZoneMod();
+	
+	if(!auto_citizenZonePrep(goal))
+	{
+		return false;
+	}
+	switch(goal)
+	{
+		case "spec": //Special handling for zones that depend on other IOTMs to be useful
+			foreach loc in specZones
+			{
+				if(!can_adventure(loc))
+				{
+					continue;
+				}
+				handleFamiliar(eagle);
+				set_property("auto_forceFreeRun", true);
+				if(autoAdv(loc))
+				{
+					activeCitZoneMod = activeCitZoneMod();
+					if(contains_text(activeCitZoneMod, goal)) //need this if statement separate in case we hit a non-combat
+					{
+						handleTracker("Citizen of a Zone: " + goal, "auto_otherstuff");
+						return true;
+					}
+					else
+					{
+						auto_log_debug("Attempted to get citizen of a zone buff for a special goal however we failed.");
+						return false;
+					}
+				}
+				else
+				{
+					auto_log_debug("Attempted to get citizen of a zone buff for a special goal however we failed.");
+					return false;
+				}
+			}
+			break;
+		case "meat": //Get +50% meat
+			foreach loc in meatZones
```

Comment:
given these entries are identical except the zone, you could extract a function taking a `boolean[location] zones`

Reply:
and then define it inside this function, like is done in providers